.github/workflows/publish_to_docker_hub.yaml

```diff
@@ -0,0 +1,106 @@
+# This workflow publishes a new docker image to 'https://hub.docker.com/r/stellar/wallet-backend'
+# when a new release is created or when we merge something to the develop branch.
+name: Docker Image Public Release
+
+on:
+  release:
+    types:
+      - published
+  push:
+    branches:
+      - develop
+jobs:
+  tests:
+    uses: ./.github/workflows/go.yaml # execute the callable go.yml
+    secrets: inherit # pass all secrets
+
+  e2e_integration_test:
+    uses: ./.github/workflows/e2e_integration_test.yaml # execute the callable e2e_integration_test.yml
+    needs:
+      - tests
+    secrets: inherit # pass all secrets
+
+  build_and_push_docker_image_on_release:
+    if: github.event_name == 'release'
+    name: Push to DockerHub (release prd) # stellar/wallet-backend:{VERSION}
+    runs-on: ubuntu-latest
+    needs:
+      - tests
+      - e2e_integration_test
+    steps:
+      - name: Check if tag is not empty
+        run: |
+          if [[ -z "${{ github.event.release.tag_name }}" ]]; then
+            echo "Release tag name cannot be empty."
+            exit 1
+          fi
+
+      - name: Determine Docker Tags
+        id: docker_tags
+        run: |
+          if [ "${{ github.event.release.prerelease }}" = "false" ]; then
+            echo "TAGS=stellar/wallet-backend:${{ github.event.release.tag_name }},stellar/wallet-backend:latest" >> $GITHUB_OUTPUT
+          else
+            echo "TAGS=stellar/wallet-backend:${{ github.event.release.tag_name }}" >> $GITHUB_OUTPUT
+          fi
+
+      - uses: actions/checkout@v4
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push to DockerHub (release prd)
+        uses: docker/build-push-action@v6
+        with:
+          push: true
+          build-args: |
+            GIT_COMMIT=${{ github.event.release.tag_name }}
+          tags: ${{ steps.docker_tags.outputs.TAGS }}
```

Comment:
You may wish to post a status to github with the tag name, so that when viewing the CI runs of a commit, you an see the tag.

We do this on the quickstart image, see it here:
https://github.com/stellar/quickstart/blob/72445f714ef16568d801e5cf617d7e086dfd48bd/.github/actions/push/action.yml#L63-L75

![gif](https://github.com/user-attachments/assets/d5a491d6-9e7b-42dc-84ed-0afbc3eda26c)

Reply:
Good idea! Addressed in https://github.com/stellar/wallet-backend/pull/169/commits/917cb2f46e19980f837c405641ca2c2d9f07511f.

Results:
<img width="772" alt="Screenshot 2025-05-06 at 11 54 24 AM" src="https://github.com/user-attachments/assets/5923b8a1-d875-4949-b2f9-e7d29c76a76b" />